system/CodeIgniter.php

```diff
@@ -528,7 +528,7 @@ protected function detectEnvironment()
     {
         // Make sure ENVIRONMENT isn't already set by other means.
         if (! defined('ENVIRONMENT')) {
-            define('ENVIRONMENT', $_SERVER['CI_ENVIRONMENT'] ?? 'production');
+            define('ENVIRONMENT', getenv('CI_ENVIRONMENT') ?? 'production');
```

Comment:
See Paul's suggestion ~~below~~ above:
```suggestion
            define('ENVIRONMENT', env('CI_ENVIRONMENT', 'production'));

```
It might seem cyclical but I checked and everything needed is loaded by this point.